src/rr/dnssec/algorithm.rs

```diff
@@ -230,7 +310,15 @@ impl Algorithm {
         bytes.extend_from_slice(&n);
 
         bytes
-      }
+      },
+      Algorithm::ECDSAP256SHA256 | Algorithm::ECDSAP384SHA384 => {
+        let pkey = if let DnssecKey::Ecdsa(ref key) = *public_key {
+            key
+        } else {
+            panic!("Invalid type of key")
+        };
+        pkey.to_wire()
```

Comment:
Do you know if the format on this is correct? I had to rip open the RSA key above to put it on (and read it from) the stream properly. The only docs I found for ECDS are not very detailed, 
https://tools.ietf.org/html/draft-hoffman-dnssec-ecdsa-04#section-4:

```
ECDSA public keys consist of a single value, called "Q" in FIPS
   186-3.  In DNSSEC keys, Q is a simple bit string that represents the
   uncompressed form of a curve point, "x | y".

   The ECDSA signature is the combination of two non-negative integers,
   called "r" and "s" in FIPS 186-3.  The two integers, each of which is
   formatted as a simple octet string, are combined into a single longer
   octet string for DNSSEC as the concatenation "r | s".

   The algorithm numbers associated with the DNSKEY and RRSIG resource records are fully defined in the IANA Considerations section.  They
   are:

   o  DNSKEY and RRSIG RRs signifying ECDSA with the P-256 curve and
      SHA-256 use the algorithm number {TBA-2}.

   o  DNSKEY and RRSIG RRs signifying ECDSA with the P-384 curve and
      SHA-384 use the algorithm number {TBA-3}.

   Conformant implementations MUST support signing and/or validation of
   signatures with both ECDSA with the P-256 curve and SHA-256, and with
   ECDSA with the P-384 curve and SHA-384.
```